src/projection.jl

```diff
@@ -165,27 +167,29 @@ end
 (::ProjectTo{T})(dx::Integer) where {T<:Complex{<:AbstractFloat}} = convert(T, dx)
 
 # Other numbers, including e.g. ForwardDiff.Dual and Symbolics.Sym, should pass through.
-# We assume (lacking evidence to the contrary) that it is the right subspace of numebers
-# The (::ProjectTo{T})(::T) method doesn't work because we are allowing a different
-# Number type that might not be a subtype of the `project_type`.
+# We assume (lacking evidence to the contrary) that it is the right subspace of numebers.
 (::ProjectTo{<:Number})(dx::Number) = dx
 
 (project::ProjectTo{<:Real})(dx::Complex) = project(real(dx))
 (project::ProjectTo{<:Complex})(dx::Real) = project(complex(dx))
 
+# Tangents: we prefer to reconstruct numbers, but only safe to try when their constructor
+# understands, including a mix of Zeros & reals. Other cases, we just let through:
+(project::ProjectTo{<:Complex})(dx::Tangent{<:Number}) = project(Complex(dx.re, dx.im))
```

Comment:
My first thought is: When do we ever see a `Tangent` of a Complex number? should it not be just the natural differential, i.e. a Complex number?

Another concern is that say `Tangent{MyNumber}` may not have `re` and `im` fields, so should we not have something like:  
```suggestion
(project::ProjectTo{<:Complex})(dx::Tangent{<:Complex}) = project(Complex(dx.re, dx.im))
```

Reply:
Good point about fields. 

You can make one, although you shouldn't:
```
julia> gradient(x -> x.re^2, 2+3im)
((re = 4, im = nothing),)
```
Here what I think should happen is the rule for getproperty should probably finish up with a projection call, which would hit this line. Xref discussion here: https://github.com/JuliaDiff/ChainRulesCore.jl/issues/441#issuecomment-900776867